src/server/pkg/storage/fileset/pg_store.go

```diff
@@ -0,0 +1,105 @@
+package fileset
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/jmoiron/sqlx"
+	"github.com/pachyderm/pachyderm/src/server/pkg/errutil"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/fileset/index"
+)
+
+var _ Store = &postgresStore{}
+
+type postgresStore struct {
+	db *sqlx.DB
+}
+
+func NewPostgresStore(db *sqlx.DB) Store {
+	return &postgresStore{db: db}
+}
+
+func (s *postgresStore) PutIndex(ctx context.Context, p string, idx *index.Index) error {
+	if idx == nil {
+		idx = &index.Index{}
+	}
+	data, err := proto.Marshal(idx)
+	if err != nil {
+		return err
+	}
+	_, err = s.db.ExecContext(ctx,
+		`INSERT INTO storage.paths (path, index_pb)
+		VALUES ($1, $2)
+		ON CONFLICT (path) DO UPDATE SET index_pb = $2
```

Comment:
Similar to uploading the same chunk, it might make more sense for this to be an error. Particularly if the pb is supposed to be immutable.